python/paddle/fluid/optimizer.py

```diff
@@ -5044,14 +5044,20 @@ def _rename_gradient_var_name(self, block):
                 new_grad_name = name + "@MERGED"
                 self._rename_arg(op, name, new_grad_name)
 
-    def _accumulate_gradients(self, block, pp_allreduce_in_optimize=False):
+    def _accumulate_gradients(self,
+                              block,
+                              pp_allreduce_in_optimize=False,
+                              fp16_allreduce=False):
         """
         Create a new merged gradient for each parameter and accumulate the
         corresponding gradient to it.
         """
         merged_gradient_names = []
         first_opt_op_idx = None
 
+        merged_suffix = '@MERGED@FP16' if fp16_allreduce else '@MERGED'
```

Comment:
should explain the suffix for gradname for later maintainer, we now have two many suffix for grad, immediately grad, accumulated grad, casted grad, etc

Reply:
ok, add in next PR